### DISCUSSION TITLE
Translating JWMMF: A new way

### DISCUSSION THREAD

**Discussion by sircharlo** [sircharlo/meeting-media-manager#298] [Translations]
So, this project has grown in size over the past years, as some of you who have been here from the start probably noticed. As such, JWMMF went from only English in the beginning, to supporting ten languages and counting! Thank you to all contributors, your help is truly appreciated. 

Now, I'm introducing something new: **translation collaboration**. To simplify the process of translating the app and all its strings, I've started implementing GitLocalize.

If any one of you would like to volunteer to become a **JWMMF language moderator** for a language you speak, please [sign into GitLocalize](https://gitlocalize.com/) using your GitHub account, and then let me know by adding a comment in this chat. 

As language moderator, you will get notified of changes to the English original strings, have a nice and easy to use interface to review existing as well as untranslated strings in your language, and submit new translations to the repo easily. 

Thanks again to all past and current contributors, and I'm looking forward to using this new method to streamline collaboration in the future.

**Comment by joaochris** [comment]
We thank you for your work and dedication! It's a pleasure to help! I was wondering if there was an easier way to get notified of string changes and you figured it out :). I am available for pt and pt-pt :D

**Reply by sircharlo** [reply]
Thank you! Did you sign in to GitLocalize using your GitHub account?

**Reply by joaochris** [reply]
Yes! :)

**Reply by sircharlo** [reply]
Added you to pt; pt-PT coming up.

**Reply by joaochris** [reply]
Thanks!

**Reply by sircharlo** [reply]
Thank you!

**Comment by sircharlo** [comment]
If you were brought here because I pinged you, then hello! I am simply checking in to see if you would be interested in what was mentioned in the first post. Please feel no pressure to participate or accept the request, I understand fully whatever you decide. Thanks for what you've done to help!

@GionatanFazio
@stepstra
@luisenriquereyes
@Hi-there-how-are-u
@wikmanipoiss
@mtdvlpr 
@joaochris

**Reply by stepstra** [reply]
Yes, available for German, thank you! Didn't notice that here before today ...

**Reply by sircharlo** [reply]
> Yes, available for German, thank you! Didn't notice that here before today ...

Thank you! Let me know when you've signed in and I'll add you.

**Reply by stepstra** [reply]
Right away! ;-)

**Reply by stepstra** [reply]
Do I have to do something else? I don't see a translation project yet ...

**Reply by sircharlo** [reply]
Odd, I don't see you in the list of users on GitLocalize. Your username is `stepstra`, correct? And you were able to sign into GitLocalize with your GitHub account successfully?

**Reply by stepstra** [reply]
yes

**Reply by stepstra** [reply]
![image](https://user-images.githubusercontent.com/10519895/167806050-1faeb97f-1f5f-41ea-a24c-3404bd368f91.png)

**Reply by riemsesy** [reply]
Stepstra sounds very Dutch :-)

**Reply by sircharlo** [reply]
How bizarre. You just showed up now. I won't ask too many questions 😂 I've added you as a German moderator. Thank you!!

**Reply by stepstra** [reply]
Maybe I had to enter spoken languages in my profile first. That's what I did at least. ;-)

**Reply by stepstra** [reply]
> Stepstra sounds very Dutch :-)

Well, it's just the initials of my full name STEPhan STRAuch.

**Reply by riemsesy** [reply]
Das ist sehr Deutsch 😊

**Comment by sircharlo** [comment]
For how to actually review and modify translations once you've become a language moderator, check out [this guide](https://docs.gitlocalize.com/how_to_contribute.html).

**Comment by mtdvlpr** [comment]
Hi, of course I want to contribute! It's my pleasure. Thank you for all the hard work! I signed up for Gitlocalize and am available for Dutch (nl) translation :)

**Reply by sircharlo** [reply]
Thank you! Added as moderator

**Reply by riemsesy** [reply]
I can be his backup :-)

**Reply by sircharlo** [reply]
I like it! @riemsesy

**Comment by gionatanfazio** [comment]
yes ok for me, thanks

Gionatan FAZIO
***@***.***


Il giorno lun 9 mag 2022 alle ore 12:57 Charles Olivier Savignac <
***@***.***> ha scritto:

**Reply by sircharlo** [reply]
Thank you! Please let me know when you've logged into GitLocalizer with your GitHub account

**Comment by wikmanipoiss** [comment]
Thank you for your hard work. I would be happy to help with estonian (et) language. I have logged into GitLocalizer :)

**Reply by sircharlo** [reply]
Thank you so much! I added you as moderator for Estonian

**Comment by Hi-there-how-are-u** [comment]
Absolutely!
Swedish for me and maybe I could do the Finnish version in the future. But just maybe... :)
I have logged into the GitLocalize.

**Reply by sircharlo** [reply]
Thanks 😄 I've added you as moderator for both, but no pressure. Any help is appreciated!

**Comment by luisenriquereyes** [comment]
I am all setup now with GitLocalize to translate for Spanish.  Please let me know when ready to do so 👍

**Reply by sircharlo** [reply]
Thank you! You have been added to Spanish in GL.

**Comment by Hi-there-how-are-u** [comment]
Is there any way of getting a notification when there is new text to translate? Now I must manually go and see if there is anything new. My brain is made of teflon so nothing sticks to it. So I tend to forget to check. Notifications are good... :)

**Reply by sircharlo** [reply]
Hmm good point. I don't think GL or even GH has anything out of the box; maybe something like [this](https://app.github-file-watcher.com)?

**Reply by Hi-there-how-are-u** [reply]
I tried and that works really well. I made it so that it will track the en.json file.

**Comment by luisenriquereyes** [comment]
For Spanish, do I just translate the locales/es.json  file only?   Or other files also need translating?   I ask because in GitLocalize, some folders and files show as incomplete status.

**Reply by sircharlo** [reply]
Which other files show up as incomplete? Do you have an example?

**Reply by luisenriquereyes** [reply]
> Which other files show up as incomplete? Do you have an example?

These folders: 
![image](https://user-images.githubusercontent.com/78366076/168613796-4b8e14d2-3967-4e4d-a4e9-8044497bf94b.png)

**Reply by sircharlo** [reply]
Hm. Didn't realize you saw that. I only see this: 
![image](https://user-images.githubusercontent.com/1275666/168616441-385621f2-9c57-4936-8275-826cd333bf7c.png)


In any case, you only need to translate what is in the locales folder, the Spanish file.

**Reply by luisenriquereyes** [reply]
> locales/es.json file

Thanks 👍

**Comment by mtdvlpr** [comment]
Dear translators, I want to call upon your willing spirits to also help translate the beta version of a new and improved M³ before its release. If you are willing, please let me know and tell me which languages you can translate. I will add you to the gitLocalize repo, the same way you are translating the current version of M³. Thank you in advance for your help!

**Reply by Hi-there-how-are-u** [reply]
Yes I willing to translate to Swedish and Finnish.

**Reply by mtdvlpr** [reply]
Thank you!

**Reply by joaochris** [reply]
I pt-br and pt-pt. Congratulations on your rich work! Another programming genius hehehe. I haven't tested your version yet but I will soon 😉

**Reply by mtdvlpr** [reply]
Thanks! Be sure to mention any feedback you have in the discussion dedicated to the new version :)

**Reply by e-Feitosa** [reply]
> Eu pt-br e pt-pt. Parabéns pelo seu rico trabalho! Outro gênio da programação hehehe. Ainda não testei sua versão, mas em breve farei. 😉

Obrigado pela sua colaboração!
O idioma pt-BR agradece.
😁

**Reply by joaochris** [reply]
> > Eu pt-br e pt-pt. Parabéns pelo seu rico trabalho! Outro gênio da programação hehehe. Ainda não testei sua versão, mas em breve farei. 😉
> 
> Obrigado pela sua colaboração! O idioma pt-BR agradece. 😁

Valeu também pelas contribuições 🧑🏻‍🚀

**Reply by mynetx** [reply]
I can translate to Pidgin West Africa (organisation code PGW). I've registered on GitLocalize. :)

**Comment by mtdvlpr** [comment]
For those of you who are unfamiliar with the new beta version. It's available at [mtdvlpr/meeting-media-manager](https://github.com/mtdvlpr/meeting-media-manager). It's currently in the beta phase and will eventually become an official release here. So in time, you will get this version automatically. If you're curious and want to test out this version before its official release, feel free!

**Comment by mtdvlpr** [comment]
Dear translators, I have refactored the documentation website to make it easier to translate. You can now translate one page at a time and you will see the entire page as a preview. This way you know the context of translations and you have more freedom with translating links and page titles. Thank you all for your hard work!
